### DISCUSSION TITLE
PeerTube-less interface to the chat

### DISCUSSION THREAD

**Discussion by Utopiah** [JohnXLivingston/peertube-plugin-livechat#108] [Ideas]
Hi, and thanks for for the great work.

I'm preparing a hybrid event where speakers will be live streamed to a physical and remote audience. The video projector will show the video and optionally the chat.

I'd like for the audience to share feedback as they watch via the chat, asking questions, making remarks, and for the speakers to see via a laptop what is happening. 

For that I'm thinking of an alternative view where only the chat is visible, not chat and video.

It can be done by hiding the video using CSS but I imagine there might be a better way. Any suggestion welcome.

**Comment by JohnXLivingston** [comment]
Actually, this feature already exists.
The easiest way: on top of the chat, on the left of the «close» button, there is a button that open the chat fullscreen:
![image](https://user-images.githubusercontent.com/38844060/195350200-50743a6e-de61-42bd-ae7e-81da29d98a11.png)
You can then share the url of the page.

There is another/better solution.
The video owner has an extra button:
![image](https://user-images.githubusercontent.com/38844060/195350549-6fa832a8-42bc-426e-a923-27ff028b874e.png)
Note: instance's admin can change the visibility of this «share button» in the plugin settings, so that any user can see it.

This button opens this popup:
![image](https://user-images.githubusercontent.com/38844060/195350756-1e2e6985-9226-4abe-9755-0e480538af36.png)

You can then copy and share this url (by default it is the same as the one from the easy solution).

You can try the fullscreen mode here:
https://videos.john-livingston.fr/plugins/livechat/router/webchat/room/c5d20adf-1088-4d40-bbfc-1abb42d20b05 (anonymous users must first choose a nickname).

Please note that with this popup, you can also check option to have a «read-only» mode that you can use in OBS for example, to include the tchat in the livestream (using a «browser source»):
![image](https://user-images.githubusercontent.com/38844060/195351922-a7596a95-3ba0-4b09-b0f5-cb7d8df41864.png)

![image](https://user-images.githubusercontent.com/38844060/195351581-648d0bf7-631d-455b-b056-d711e27afce6.png)

Note: in the OBS source properties, you can change the background transparency using CSS:
```css
:root {--livechat-transparent: rgba(255 255 255 / 50%) !important;}
```

PS: the share popup is only available if the plugin is running in the recommended mode: «Prosody server controlled by Peertube». Other modes are deprecated.

**Comment by JohnXLivingston** [comment]
You can try the share popup here, I enabled it for all users:
https://videos.john-livingston.fr/w/qqNQwFDerUB1KnFA6N6Jbi

**Comment by Utopiah** [comment]
Amazing, thanks a lot for the clarification and the prompt reply!

**Comment by JohnXLivingston** [comment]
You are welcome :)

**Comment by JohnXLivingston** [comment]
@Utopiah , I forgot: this is not well documented for now, but you can also customize the colors.

In the modal, check «use current theme color». You will get this url:
«https://videos.john-livingston.fr/plugins/livechat/router/webchat/room/c5d20adf-1088-4d40-bbfc-1abb42d20b05?_ac_mainForeground=%23212529&_ac_mainBackground=%23fff&_ac_greyForeground=%23585858&_ac_greyBackground=%23E5E5E5&_ac_menuForeground=%23fff&_ac_menuBackground=%23000&_ac_inputForeground=%23212529&_ac_inputBackground=%23fff&_ac_buttonForeground=rgb%28255%2C+255%2C+255%29&_ac_buttonBackground=hsl%2824deg%2C+90%25%2C+50%25%29&_ac_link=%23212529&_ac_linkHover=%23212529&_readonly=noscroll&_transparent=true»

Here, they are parameters for the colors to use. You can change values by replacing them by any CSS valid color (correctly URI escaped).

**Comment by JohnXLivingston** [comment]
For example (I just replaced the «mainForeground» by `red`:
https://videos.john-livingston.fr/plugins/livechat/router/webchat/room/c5d20adf-1088-4d40-bbfc-1abb42d20b05?_ac_mainForeground=red&_ac_mainBackground=%23fff&_ac_greyForeground=%23585858&_ac_greyBackground=%23E5E5E5&_ac_menuForeground=%23fff&_ac_menuBackground=%23000&_ac_inputForeground=%23212529&_ac_inputBackground=%23fff&_ac_buttonForeground=rgb%28255%2C+255%2C+255%29&_ac_buttonBackground=hsl%2824deg%2C+90%25%2C+50%25%29&_ac_link=%23212529&_ac_linkHover=%23212529&_readonly=noscroll&_transparent=true
